src/pool.c

```diff
@@ -60,14 +76,16 @@ static int __pool_del(int n)
    struct context *ctx;
    int i;
 
+   BUG_ON(n > poolsz);
```

Comment:
`BUG_ON()` should be called when we can't recover.
In this case, and below, we can.

Returning an error code would be ok.